### PR TITLE
Add SystemOptional type for Presence

### PR DESCRIPTION
**Pull request by yongchongye**
Split `System` type into 
- `SystemRequired` 
- `SystemOptional`

### CODE REVIEW

File: src/Main.elm [hunk 1]
```diff
@@ -112,10 +112,15 @@ stringFromViewMode viewMode =
...
 
 
+type alias System =
+    { name : String, description : String }
+
+
 type Presence
     = Required
     | Optional
-    | System { name : String, description : String }
```
Comment (alvinchoong) [inline src/Main.elm:118]: we should probably retain the `System` too for backwards compatibility?

File: src/Main.elm [hunk 2]
```diff
@@ -1370,8 +1401,13 @@ decodePresence =
             |> Json.Decode.andThen
                 (\type_ ->
                     case type_ of
-                        "System" ->
-                            Json.Decode.succeed (\name description -> System { name = name, description = description })
```
Comment (choonkeat-govtech) [inline src/Main.elm:1374]: If we can decode string `"system"` as `SystemRequired` then we’re backward compatible with existing json

File: src/Main.elm (hunk 1 above)
Comment (choonkeat-govtech) [reply src/Main.elm:118]: we can update decoder to achieve backward compatibility https://github.com/choonkeat/tiny-form-fields/pull/1/files#r1729699648

new jsons encoded won’t have `"System"` anymore.

File: src/Main.elm (hunk 2 above)
Comment (yongchongye) [reply src/Main.elm:1374]: updated to handle `"System"` as `SystemRequired`

https://github.com/choonkeat/tiny-form-fields/pull/1/files#diff-2dd82f159d96fbfcd26fb7d885d25e0d54efde9e19a42494b416fa84a5aca568R1405

**Comment by choonkeat-govtech** [comment]
🙏

**Pull request merged**